### DISCUSSION TITLE
Isolated template components

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#164] [Ideas]
While the `include` template tag is useful for some scenarios, a different approach is needed for actual applications.
I recommend adding a `component` tag.
Differences from `include`:
- Components can not access variables in the global template scope, only variables provided by context producers.
- Components have to specify which parameters they need (and whether a given parameter is required or optional with a default value).
- Instantiating a component without a required parameter would throw error.
- Components can have slots.

**Comment by treagod** [comment]
I like the idea.

What would be interesting is how and where to define the component and how to include the component. Using a path to the component file seems odd to me. Possible solution could be to have a component directory from where Marten loads the components in a registry with a name as their key.

Then you could load the component by refering to its name. 

I created #165 for discussion
